### PR TITLE
Backports from msys2-runtime

### PR DESCRIPTION
**Pull request by dscho**
Actually, I wanted to see whether the patches apply cleanly, and they do. I did not have time to test yet, though, but feel free to merge if you can verify that the changes are sound.

### CODE REVIEW

File: src/path_conv.cpp [hunk 1]
```diff
@@ -587,6 +589,11 @@ void ppl_convert(const char** from, const char* to, char** dst, const char* dste
             subp_convert(&beg, it, is_url, dst, dstend);
             is_url = 0;
 
+            if (*dst == dstend) {
+                system_printf("Path cut off during conversion: %s\n", orig_dst);
```
Comment (Alexpux) [inline src/path_conv.cpp]: there are no `system_printf` function

Comment (dscho) [inline src/path_conv.cpp]: Sorry! Fixed.

**Comment by dscho** [comment]
Thanks!

**Comment by Alexpux** [comment]
I'm not test it. Just merge

**Comment by dscho** [comment]
> I'm not test it. Just merge

You did not test it before merging? Whoa ;-)

In any case, I tested and it works as intended.

**Comment by mingwandroid** [comment]
AFAIK, I tested these changes before merging them to msys2-runtime and told
Alexey about this so it's not such a big deal!

**Pull request merged**